public/app/features/datasources/settings/HttpSettingsCtrl.ts

```diff
@@ -24,6 +24,27 @@ coreModule.directive('datasourceHttpSettings', () => {
 
         $scope.onBasicAuthPasswordReset = createResetHandler($scope, PasswordFieldEnum.BasicAuthPassword);
         $scope.onBasicAuthPasswordChange = createChangeHandler($scope, PasswordFieldEnum.BasicAuthPassword);
+
+        $scope.onHeadersChange = headers => {
+          $scope.current.secureJsonData = $scope.current.secureJsonData || {};
+          for (const [index, header] of headers.entries()) {
+            $scope.current.jsonData[`httpHeaderName${index + 1}`] = header.name;
+            if (!header.configured) {
+              $scope.current.secureJsonData[`httpHeaderValue${index + 1}`] = header.value;
+            }
+          }
+
+          Object.keys($scope.current.jsonData)
+            .sort()
```

Comment:
Any particular reason for sorting here? Header names are added with incremental index, so when retrieving the keys from `jsonData`, the ones that you are interested in should be sorted already.